includes/emails/class-sensei-email-sender.php

```diff
@@ -0,0 +1,112 @@
+<?php
+/**
+ * File containing the Sensei_Email_Sender class.
+ *
+ * @package sensei
+ */
+
+// phpcs:disable VariableAnalysis.CodeAnalysis.VariableAnalysis -- Prevent "Unused global variable $sensei_email_data"
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+require plugin_dir_path( __DIR__ ) . '../vendor/autoload.php';
+
+use \Pelago\Emogrifier\CssInliner;
+
+/**
+ * Class Sensei_Email_Sender
+ *
+ * @package Sensei\Internal\Emails
+ */
+class Sensei_Email_Sender {
+
+	/**
+	 * The constructor.
+	 *
+	 * @var array
+	 */
+	public function __construct() {
+		add_action( 'post_updated', [ $this, 'send_email_from_email_post' ], 10, 2 );
+	}
+
+
+	/**
+	 * Check if the post is an email post and send the email.
+	 *
+	 * @param int     $post_ID The post ID.
+	 * @param WP_Post $post    The post object.
+	 */
+	public function send_email_from_email_post( $post_ID, $post ) {
+
+		if ( \Sensei\Internal\Emails\Email_Post_Type::POST_TYPE !== $post->post_type ) {
+			return;
+		}
+
+		$templated_output = $this->get_templated_post_content( $post );
+
+		$style_string = apply_filters( 'sensei_email_styles', $this->get_header_styles() );
+
+		$html_output_with_inlined_css = CssInliner::fromHtml( $templated_output )->inlineCss( $style_string )->render();
+
+		$headers[] = 'Content-Type: text/html; charset=UTF-8';
+
+		wp_mail( apply_filters( 'sensei_test_email_address', 'test-email@senseitestemailcheck.com' ), 'Dummy text', $html_output_with_inlined_css, $headers, null );
```

Comment:
As per our [hook naming conventions](https://senseip2.wordpress.com/programming-standards-best-practices/naming-conventions/#hooks), this should probably be called something like `sensei_email_address`.